src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/QueryHelper.scala

```diff
@@ -62,16 +54,20 @@ private[querygeneration] case class QueryHelper(
   val processedProjections: Option[Seq[NamedExpression]] = projections
     .map(
       p =>
-        p.map(
-          e =>
+        p.map {
+          case alias: Alias =>
+            println("Hit Alias: " + alias.getClass.getCanonicalName)
+            alias
+          case e =>
+            println("Hit Attribute: " + e.getClass.getCanonicalName)
             colSet.find(c => c.exprId == e.exprId) match {
               case Some(a) =>
                 AttributeReference(a.name, a.dataType, a.nullable, a.metadata)(
                   a.exprId
                 )
               case None => e
-          }
-      )
+            }
+        }
     )
     .map(p => renameColumns(p, alias))
```

Comment:
Below is the code from spark, it looks it used the child's expr id. `Alias(new First(attr).toAggregateExpression(), attr.name)(attr.exprId)`
```
/**
 * Replaces logical [[Deduplicate]] operator with an [[Aggregate]] operator.
 */
object ReplaceDeduplicateWithAggregate extends Rule[LogicalPlan] {
  def apply(plan: LogicalPlan): LogicalPlan = plan transform {
    case Deduplicate(keys, child) if !child.isStreaming =>
      val keyExprIds = keys.map(_.exprId)
      val aggCols = child.output.map { attr =>
        if (keyExprIds.contains(attr.exprId)) {
          attr
        } else {
          Alias(new First(attr).toAggregateExpression(), attr.name)(attr.exprId)
        }
      }
      // SPARK-22951: Physical aggregate operators distinguishes global aggregation and grouping
      // aggregations by checking the number of grouping keys. The key difference here is that a
      // global aggregation always returns at least one row even if there are no input rows. Here
      // we append a literal when the grouping key list is empty so that the result aggregate
      // operator is properly treated as a grouping aggregation.
      val nonemptyKeys = if (keys.isEmpty) Literal(1) :: Nil else keys
      Aggregate(nonemptyKeys, aggCols, child)
  }
}
```